File: January 9, 2026.js
function isCircularPrime(n) {

  const str = String(n);

  for (let i = 0; i < str.length; i++) {
    let rotationN = Number(str.slice(1) + str[0]);
    for (let j = 2; j <= Math.floor(Math.sqrt(rotationN)); j++) {
      if (rotationN % j == 0) {
        return false;
      }
    }
  }

  return true;
}

File: January 9, 2026.py
def is_circular_prime(n):

    s = str(n)

    for i in range(len(s)):
        rotation_n = int(s[1:] + s[0])
        for j in range(2, int(rotation_n**0.5) + 1):
            if rotation_n % j == 0:
                return False

    return True